nodes/src/nodes/remote/client/IGlobal.py

```
@@ -67,21 +68,30 @@ def beginGlobal(self):
+        safe_control_url, host_headers = build_ssrf_safe_url(self.urlControl, hostname, resolved_ips)
+        # Merge Host header into request headers for all subsequent calls
+        self.headers.update(host_headers)
+        # Store the IP-pinned base for use in all requests
+        self._safe_control_url = safe_control_url
+        safe_use_url, _ = build_ssrf_safe_url(f'http://{self._host}:{self._port}/use?name=remote', hostname, resolved_ips)
+
         # Update the monitor
         monitorStatus('Loading remote pipe')
 
-        requests.post(f'http://{self._host}:{self._port}/use?name=remote', headers=self.headers).raise_for_status()
+        requests.post(safe_use_url, headers=self.headers, allow_redirects=False).raise_for_status()
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)